packages/x-data-grid/src/components/GridPagination.tsx

```diff
@@ -28,6 +30,25 @@ const GridPaginationRoot = styled(TablePagination)(({ theme }) => ({
   },
 })) as typeof TablePagination;
 
+export type WrappedLabelDisplayedRows = (
+  args: LabelDisplayedRowsArgs & { estimated?: number },
+) => React.ReactNode;
+
+const wrapLabelDisplayedRows = (
+  labelDisplayedRows: WrappedLabelDisplayedRows,
+  estimated?: number,
+): TablePaginationProps['labelDisplayedRows'] => {
+  return ({ from, to, count, page }: LabelDisplayedRowsArgs) =>
+    labelDisplayedRows({ from, to, count, page, estimated });
+};
+
+const defaultLabelDisplayedRows: WrappedLabelDisplayedRows = ({ from, to, count, estimated }) => {
+  if (!estimated) {
+    return `${from}–${to} of ${count !== -1 ? count : `more than ${to}`}`;
```

Comment:
Shouldn't this sentence be in the localization?
Even if people can override it, that way people just have to pass there usual locale (which they probably already do) and they have a translated app.

Reply:
Yes, ideally `estimated` should be supported by the `@mui/material/TablePagination` (since that's where its localization is coming from). It would also mean a minimum peer dependency bump apart from adding the support in that component.
Until feasible, we can provide this workaround to our users to achieve the localization for the estimated row count use case. See the info callout just above [this](https://deploy-preview-12474--material-ui-x.netlify.app/x/react-data-grid/pagination/#cursor-based-pagination) docs section.

Or your suggestion is more about moving localization for this function (`labelDisplayedRows`) to the Grid localization already?

Reply:
I was thinking about the grid localization, but it's true that it could also belong to the `TablePagination` one...
IMHO if we plan to put it in `TablePagination`, then we could create the PR in the core repo to make sure that we have it for our next major.